Fix regex sets pod to handle more /flags, and update pod

khwilliamson opened the pull request:
The pod didn't reflect restrictions on what can go into such a class.

khwilliamson committed:
perlrecharclass: Update regex sets pod

The pod didn't reflect restrictions on what can go into such a class.

jkeenan commented:
> The pod didn't reflect restrictions on what can go into such a class.

Can you point us to places in the test suite which exercise the code whose documentation needs adjusting?

khwilliamson commented:
On 2/24/22 13:34, James E Keenan wrote:
>     The pod didn't reflect restrictions on what can go into such a class.
> 
> Can you point us to places in the test suite which exercise the code 
> whose documentation needs adjusting?
> 

t/re /reg_mesg.t
      /regex_sets.t
      /regex_sets_compat.t

khwilliamson committed:
regex_sets: Allow all legal flags in nested calls

See https://github.com/Perl/perl5/issues/16779#issuecomment-563219424

This commit extends the flags accepted by nested calls to regex sets to
any legal set.  Previously it allowed only '^', as that is what an
actual compilation would return.  But as pointed out in the conversation
in that ticket, it's possible to detach flags from the rest of the
pattern, and to write equivalent ones in multiple ways.

This commit just changes one line to look for any legal flag, not
restricting it to '^'.
